### ISSUE TITLE
Fragmentation strategy to segment (sub)datasets (or pods) in one file

### ISSUE THREAD

**Issue by constraintAutomaton** [SolidBench/rdf-dataset-fragmenter.js#41]
Hi, I am wondering if there is currently a way (or a straightforward way to be more precise, because I think this **might** be possible by playing with the regex of the quad sink) to make it so that each dataset (or pod in the context of Solidbench) has only one file containing all the associated data. I think @surilindur  might have done something similar.

**Comment by rubensworks** [comment]
Do you have a config for this @surilindur?

**Comment by surilindur** [comment]
Yes, but it requires some modifications to the quad sink (in a pending PR that I need to update) and also the work I did to convert `QuadMatcherPredicate` into a generic `QuadMatcherComponent` that allows matching any component with a deterministic percentage chance.

The approach I have also does not work with blank nodes, so it will need some rethinking.

**Comment by constraintAutomaton** [comment]
I think a simple quad sink could do the work where we place in the config the IRI pattern of the dataset and a template of the expected output like

```
{
"dataset": "^(.*\\/pods\\/[0-9]+\\/)",
"redirection": "$1/pod_data"
}
```

The sink would simply deviate the iri into one source by dataset.

**Comment by constraintAutomaton** [comment]
The PR #39 might solve the problem.

**Comment by surilindur** [comment]
I hope it will. It will allow, for example, the following types of mappings:
```json
"iriToPath": {
  "^http://": "out-fragments/http/",
  "^http://localhost:3000/dbpedia.org/.*": "out-fragments/http/localhost_3000/dbpedia",
  "^https://": "out-fragments/https/"
}
```
That will place all the DBPedia data into a single file at `out-fragments/http/localhost_3000/dbpedia`. The `.*` at the end is important, so it matches and replaces the entire URI and maps it directly to the given path.

**Comment by surilindur** [comment]
As another example, mapping like this will place every pod's data into a single `data` file in the pod folder:
```json
"^http://localhost:3000/pods/([0-9]+)/.*": "out-fragments/http/localhost_3000/pods/$1/data",
```
The percentage chance needs to be implemented elsewhere, though, in a quad matcher. I will make a PR for it later. It does not feel like something that would belong in a sink directly.
